src/core/Router.js

```diff
@@ -34,23 +35,27 @@ class Router {
     requestTypes = requestTypes.constructor !== Array ? [requestTypes] : requestTypes;
     protocols = protocols.constructor !== Array ? [protocols] : protocols;
 
+    let interceptor = new RedirectInterceptor();
     let page = new Page(routeId, requestTypes, protocols, route, domains);
     let routerArguments = [];
     routerArguments.push(route);
+    routerArguments.push(interceptor.patch);
     routerArguments.push((req, res, next) => {
       req.currentPage = page;
       next();
     });
-    page.getRequestFilters().forEach(b => {
+    page.filters.forEach(b => {
       let filter = this._filterFactory.getFilter(b);
-      routerArguments.push(filter.onFilter.bind(filter));
+      routerArguments.push(filter.onRequest.bind(filter));
     });
     routerArguments.push(page.render.bind(page));
-    page.getResponseFilters().forEach(a => {
+    page.filters.forEach(a => {
```

Comment:
The filter order in our internal java library given the config:
```
@FilterA
@FilterB
```
works in this order:
1. FilterA.onRequest
2. FilterB.onRequest
3. FilterB.onResponse
4. FilterA.onResponse

That's the reason why I force the user to create 2 separate filter so the execution order becomes more clear, or did I miss something?